src/ray/raylet/node_manager.cc

```diff
@@ -619,17 +619,20 @@ void NodeManager::AssignTask(Task &task) {
       // Extend the frontier to include the executing task.
       auto actor_entry = actor_registry_.find(spec.ActorId());
       RAY_CHECK(actor_entry != actor_registry_.end());
-      actor_entry->second.ExtendFrontier(spec.ActorHandleId(), spec.ActorDummyObject());
       // Update the task's execution dependencies to reflect the actual
       // execution order, to support deterministic reconstruction.
       // NOTE(swang): The update of an actor task's execution dependencies is
       // performed asynchronously. This means that if this node manager dies,
       // we may lose updates that are in flight to the task table. We only
       // guarantee deterministic reconstruction ordering for tasks whose
       // updates are reflected in the task table.
-      TaskExecutionSpecification &mutable_spec = task.GetTaskExecutionSpec();
-      mutable_spec.SetExecutionDependencies(
-          {actor_entry->second.GetExecutionDependency()});
+      auto execution_dependency = actor_entry->second.GetExecutionDependency();
+      if (!execution_dependency.is_nil()) {
```

Comment:
`execution_dependency.is_nil()` is true only for the very first actor task that is assigned, right?

Reply:
Yup.